compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp

```diff
@@ -113,6 +114,11 @@ extern "C" void LLVMRustPrintPassTimings() {
   TimerGroup::printAll(OS);
 }
 
+extern "C" void LLVMRustPrintStatistics() {
+  raw_fd_ostream OS(2, false); // stderr.
```

Comment:
Wouldn’t it be better to return a string from this function and leave it up to rustc to actually print? Sure it’ll allocate the stats into a memory buffer, but that seems like a negligible cost.

Reply:
The same is true for LLVMRustPrintPassTimings I guess, but that's preexisting.

What would be the benefit? Better reusability?

Reply:
Well, one thing I’m thinking of is that just dumping stuff onto stderr is not a great experience with output formats like JSON. It just seems that leaving outputting to rustc is just gonna leave less room for things to go very wrong in general. While I’m not anticipating this getting fixed anytime soon, but having this just output a string seems like a meaningful intermediate step.

But also, looking at that magic number makes my eyeballs really itchy, even despite the comment :)

Reply:
@nagisa I'm not opposed to doing it that way (allocating into a string). Do you want me to change `LLVMRustPrintPassTimings` too?

Reply:
That would be quite awesome!